backend/inspirehep/pidstore/providers/cnum.py

```diff
@@ -50,15 +52,30 @@ def next(cls, data):
         opening_date = data.get("opening_date")
         if not opening_date:
             return
-
         adjusted_date = datetime.strptime(opening_date, "%Y-%m-%d").strftime("%y-%m-%d")
         cnum = f"C{adjusted_date}"
-        cnums_count = PersistentIdentifier.query\
-            .filter(PersistentIdentifier.pid_value.like(f"{cnum}%"))\
-            .filter(PersistentIdentifier.pid_type == "cnum")\
-            .count()
+        all_postfixes = []
+        all_cnums = (
+            PersistentIdentifier.query.with_for_update()
+            .filter(PersistentIdentifier.pid_value.like(f"{cnum}%"))
+            .filter(PersistentIdentifier.pid_type == "cnum")
+            .all()
+        )
+        if not all_cnums:
+            postfix = None
```

Comment:
```suggestion
            return cnum
```
so you don't have to do the test on L76.

Reply:
:+1: